iree/hal/cts/BUILD

```diff
@@ -32,8 +32,8 @@ cc_library(
     ],
     deps = [
         "//iree/base:status",
-        "//iree/base:status_matchers",
         "//iree/hal:driver_registry",
+        "//iree/testing:gtest",
```

Comment:
This is already included in PLATFORM_VULKAN_TEST_DEPS, it should not be included twice:

https://github.com/google/iree/blob/6132bb38e63faefc6a244b938174db2f92fd1291/iree/build_defs.oss.bzl#L50-L53

Reply:
discussed offline: we'll need to clean this up - that is really nonsensical
ideally we'd remove PLATFORM_VULKAN_TEST_DEPS entirely, and if not entirely at least allow //iree/testing:gtest to still be taken as a dep -- that someone somewhere in the transitive deps uses vulkan shouldn't impact whether unrelated code can depend on gtest at all :(